.github/workflows/cd-api-infra.yml

```diff
@@ -0,0 +1,61 @@
+name: Deploy API Infra
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "infra/api/**"
+
+jobs:
+  build-repository:
+    name: Build Repository
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: 1.7.1
+          terraform_wrapper: false
+
+      - name: Configure AWS credentials
+        uses: ./.github/actions/configure-aws-credentials
+        with:
+          app_name: analytics
+          environment: shared
+      - name: Build Repository
+        run: |
+          ./bin/terraform-init-and-apply.sh infra/api/build-repository shared
+
+  deploy-infra:
+    name: Deploy Infrastructure
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        directory: ["database", "service"]
+        envs: [ "staging", "prod"]
```

Comment:
Do we want to hardcode the envs here? I would suggest that the caller should supply the env to deploy.

Reply:
Are you suggesting something like inputs provided in a dropdown a la  `workflow_dispatch` or something else